### ISSUE TITLE
Add support for non-json bodies in Batch content

### ISSUE THREAD

**Issue by darrelmiller** [microsoftgraph/msgraph-sdk-dotnet-core#22]
This is a missing capability in our Batch content support.  The details are [here](https://docs.oasis-open.org/odata/odata-json-format/v4.01/csprd05/odata-json-format-v4.01-csprd05.html#sec_BatchRequest) in the OData spec.

> For media type application/json or one of its subtypes, optionally with format parameters, the value of body is JSON.
> 
> For media types of top-level type text, for example text/plain, the value of body is a string containing the value of the request body.
> 
> For all other media types the value of body is a string containing the base64url-encoded value of the request body. In this case the body content can be compressed or chunked if this is correctly reflected in the Transfer-Encoding header.
> 

This needs to be fixed here: https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/dev/src/Microsoft.Graph.Core/Requests/Content/BatchRequestContent.cs#L147  for requests and we will also need to fix this for responses.  The rules for `body` are the same for responses as requests.

 
[AB#7193](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/7193)

**Comment by MIchaelMainer** [comment]
Might as well add this check since it is missing from batch creation. 
> A body MUST NOT be specified if the method is get or delete.

**Comment by mbratukha** [comment]
Hello together,

we are currently truing to use the batch request capability to fetch photos of multiple users at once.
Unfortunately, we have faced this issue, that the content type of every image is always `application/json`.

I've investigated the internals of the SDK, and have found that it happens here:
https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/6ec55dcc43b884cec3d61aeb3ceb69fe27852157/src/Microsoft.Graph.Core/Requests/Content/BatchResponseContent.cs#L197


Basically, in the line `208` the content type is hard-coded to be always `application/json`, but not overridden by the values from the original response item in the lines `211-217`:
![image](https://user-images.githubusercontent.com/1413357/154030274-7a6b74e5-c587-46f1-9e84-f2d57354df83.png)

Is there any chance that it will be fixed soon?

**Comment by mzhukovs** [comment]
I take it no updates on this? I have the same use case as most running into this - user photos by batch.

**Comment by andrueastman** [comment]
Hey @mbratukha and @mzhukovs.

Just to confirm, when you attempt to request user photos, do you run into any issues when you try to do it in this fashion?

```cs
var requestUserPhotoFile = client.Users["{My_User_Id}"].Photo.Content.Request();
var batchRequestContent = new BatchRequestContent();
var userPhotoFileRequestId = batchRequestContent.AddBatchRequestStep(requestUserPhotoFile); // you can add more photo requests here
BatchResponseContent batchResponse = await client.Batch.Request().PostAsync(batchRequestContent);
using var userPhotoStream = await batchResponse.GetResponseStreamByIdAsync(userPhotoFileRequestId );
```

**Comment by mbratukha** [comment]
He @andrueastman. exactly, our code looks pretty similar, nut we do not use the `GetResponseStreamByIdAsync` method.
Here is a simplified version of what we are trying to reach:

``` C#

var requestMessage1 = client.Users[userId1].Photo.Content.Request().GetHttpRequestMessage();
var requestMessage2 = client.Users[userId2].Photo.Content.Request().GetHttpRequestMessage();

var batchRequestContent = new BatchRequestContent();

var requestId1 = batchRequestContent.AddBatchRequestStep(requestMessage1);
var requestId2 = batchRequestContent.AddBatchRequestStep(requestMessage2);

var batchResponse = await client.Batch.Request().PostAsync(batchRequestContent);

var response1 = await batchResponse.GetResponseByIdAsync(requestId1);
var response2 = await batchResponse.GetResponseByIdAsync(requestId2);

var userPhoto1 = await response1.Content.ReadAsStreamAsync();
var mediaType1 = response1.Headers.ContentType.MediaType; // BUG: here should be 'image/jpeg' but we are getting 'application/json'

var userPhoto2 = await response2.Content.ReadAsStreamAsync();
var mediaType2 = response2.Headers.ContentType.MediaType; // BUG: here should be 'image/jpeg' but we are getting 'application/json'

```

We have traced (with Fiddler) the bulk request/response and we can see that a right response is coming from MS Graph, but it's not properly converted to the `HttResponseMessage` object.

**Comment by andrueastman** [comment]
Thanks for the feedback @mbratukha.

If I understand you correctly, you can obtain the stream correctly. However, the issue is that you have an incorrect `ContentType` header. Yes?

**Comment by mbratukha** [comment]
@andrueastman exactly.

**Comment by andrueastman** [comment]
Thanks for helping me understand the issue @mbratukha. 
I have created #368 to resolve this and you should be able to use the upcoming version 2.0.8 with the fix later in the week.

**Comment by mbratukha** [comment]
@andrueastman thank you for the quick fix!
